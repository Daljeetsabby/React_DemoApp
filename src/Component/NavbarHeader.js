import React from 'react';
import { BrowserRouter as Router, Route, Redirect, Switch, NavLink } from 'react-router-dom';


const NavbarHeader = () => {
  const loggedIn = localStorage.getItem("loggedIn");
  const user = JSON.parse(localStorage.getItem('profile'))
  function logout(id) {
    localStorage.clear();
  }

  return (
    <nav class="navbar navbar-expand-lg navbar-dark bg-primary fixed-top">
      <div className="container headercss ">
        <NavLink to="/" className="navbar-brand">Product Inventory</NavLink>
        <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
          <span class="navbar-toggler-icon"></span>
        </button>
        <div class="collapse navbar-collapse " id="navbarSupportedContent">
          <ul class="navbar-nav ml-auto">
            <li class="nav-item ">
              {!loggedIn && <NavLink className="nav-link" to="/login">login</NavLink>}
            </li>
            <li class="nav-item">
              <NavLink className="nav-link " to="/about">About</NavLink>
            </li>
            <li class="nav-item">
              {<NavLink className="nav-link" to={loggedIn ? "/addproduct" : "/login"}>Add Product</NavLink>}
            </li>
            {loggedIn && <li class="nav-item dropdown">
              <a class="nav-link dropdown-toggle" href="#" id="navbarDropdownMenuLink" role="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                {loggedIn && user ? user[0]?.emailId : ""}
              </a>
              <div class="dropdown-menu" aria-labelledby="navbarDropdownMenuLink">
                {loggedIn && <NavLink className="dropdown-item bg-white text-primary" to={loggedIn ? "/UserDetail" : "/login"} >Registration Details</NavLink>}
                <div> {loggedIn && <NavLink className=" dropdown-item bg-white text-primary " onClick={() => logout()} to="/" >Sign out</NavLink>} </div>
              </div>
            </li>
            }
          </ul>
        </div>
      </div>
    </nav>
  );

}



export default NavbarHeader;
