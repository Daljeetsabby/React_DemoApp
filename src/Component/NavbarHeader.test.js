import React from 'react';
import { shallow } from 'enzyme';
import NavbarHeader from './NavbarHeader'


describe('Navbar', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<NavbarHeader/>);
    });

    test('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});

describe('Navbar Render correctly ', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<NavbarHeader />);
    })

    it('should have Single button for Logo', () => {
        expect(wrapper.find('button')).toHaveLength(1);
    })
    it('should have Single unorder list', () => {
        expect(wrapper.find('ul')).toHaveLength(1);
    })
    it('should have three list element', () => {
        expect(wrapper.find('li')).toHaveLength(3);
    })
    it('should have 6 NavLink for routing', () => {
        expect(wrapper.find('NavLink')).toHaveLength(4);
    })

    it('Should have nav items of home and sign out', () => {
        expect(wrapper.find('nav').length).toEqual(1);
    });

    it('Test click event', () => {
       
        const button = shallow((<NavbarHeader />));
        button.find('button').simulate('click');
       
      });

})

