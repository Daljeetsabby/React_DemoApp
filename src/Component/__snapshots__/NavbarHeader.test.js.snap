// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<nav
  class="navbar navbar-expand-lg navbar-dark bg-primary fixed-top"
>
  <div
    className="container headercss "
  >
    <NavLink
      className="navbar-brand"
      to="/"
    >
      Product Inventory
    </NavLink>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      class="navbar-toggler"
      data-target="#navbarSupportedContent"
      data-toggle="collapse"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="collapse navbar-collapse "
      id="navbarSupportedContent"
    >
      <ul
        class="navbar-nav ml-auto"
      >
        <li
          class="nav-item "
        >
          <NavLink
            className="nav-link"
            to="/login"
          >
            login
          </NavLink>
        </li>
        <li
          class="nav-item"
        >
          <NavLink
            className="nav-link "
            to="/about"
          >
            About
          </NavLink>
        </li>
        <li
          class="nav-item"
        >
          <NavLink
            className="nav-link"
            to="/login"
          >
            Add Product
          </NavLink>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
